RCS file: /pack/cvsroots/wxwindows/wxWindows/src/msw/filedlg.cpp,v
retrieving revision 1.63
diff -b -u -2 -r1.63 filedlg.cpp
--- filedlg.cpp 2004/04/21 21:13:45 1.63
+++ filedlg.cpp 2004/05/11 11:51:38
@@ -274,4 +274,21 @@
     of.lpstrFile = fileNameBuffer;  // holds returned filename
     of.nMaxFile  = wxMAXPATH;
+
+    //=== Setting the default extension >>=================================
+    wxString defextBuffer;
+    if (m_dialogStyle & wxSAVE)
+    {
+        const wxChar* extension = filterBuffer;
+        int   maxFilter = (int)(of.nFilterIndex*2L) - 1;
+
+        for( int i = 0; i < maxFilter; i++ )           // get extension
+            extension = extension + wxStrlen( extension ) + 1;
+        // use dummy name a to avoid assert in AppendExtension
+        defextBuffer = AppendExtension(wxT("a"), extension);
+        if (defextBuffer.StartsWith(wxT("a.")))
+        {   defextBuffer.Mid(2);
+            of.lpstrDefExt       = defextBuffer.c_str();
+        }
+    }
 
     //== Execute FileDialog >>=================================================


 	  	 
